leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/LeshanClientBuilder.java

```diff
@@ -309,7 +320,7 @@ protected Connector createSecuredConnector(DtlsConnectorConfig dtlsConfig) {
         }
 
         return createLeshanClient(endpoint, localAddress, objectEnablers, coapConfig, dtlsConfigBuilder,
-                endpointFactory, engineFactory, additionalAttributes, encoder, decoder, executor);
+                this.trustStore, endpointFactory, engineFactory, additionalAttributes, encoder, decoder, executor);
```

Comment:
why not setting truststore in dtlsConfigBuilder ?

Reply:
We kind are -- but is is just a bit delayed.

We do configure newBuilder.setCertificateVerifier() latter which needs to have customized trust store.

And in setCertificateVerifier there is enforcement that config.trustStore must be null.

Check certificate usage code changes on how trust store gets customized.

Reply:
I missed that once truststore is set in dtlsConfig we can not set it to null anymore, so it's imposible to "clean" the dtlsConfig afterwards.

I create an issue at californium to discuss about this : https://github.com/eclipse/californium/issues/1427

I suppose at short term we can go for your solution.